src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -0,0 +1,326 @@
+        application_logs = ApplicationLogsConfig(fs_log)
+
+    http_logs = None
+    if web_server_logging:
+        enabled = (web_server_logging != 'off')
+        #100 mb max log size, retenting last 3 days. Yes we hard code it, portal does too
+        fs_server_log = FileSystemHttpLogsConfig(100, 3, enabled)
+        http_logs = HttpLogsConfig(fs_server_log)
+
+    detailed_error_messages_logs = (None if detailed_error_messages is None
+                                    else EnabledConfig(detailed_error_messages == 'on'))
+    failed_request_tracing_logs = (None if failed_request_tracing is None
+                                   else EnabledConfig(failed_request_tracing == 'on'))
+    site_log_config = SiteLogsConfig(location,
+                                     application_logs=application_logs,
+                                     http_logs=http_logs,
+                                     failed_requests_tracing=failed_request_tracing_logs,
+                                     detailed_error_messages=detailed_error_messages_logs)
+    if slot is None:
+        return client.sites.update_site_logs_config(resource_group, name, site_log_config)
+    else:
+        return client.sites.update_site_logs_config_slot(resource_group, name,
+                                                         site_log_config, slot)
+
+def config_slot_auto_swap(resource_group, webapp, slot, auto_swap_slot=None, disable=None):
+    client = web_client_factory()
+    site_config = client.sites.get_site_config_slot(resource_group, webapp, slot)
+    site_config.auto_swap_slot_name = '' if disable else (auto_swap_slot or 'production')
+    return client.sites.update_site_config_slot(resource_group, webapp, site_config, slot)
+
+def get_streaming_log(resource_group, name, provider=None, slot=None):
+    client = web_client_factory()
+    repo_url = _get_repo_url(client, resource_group, name, slot)
+    streaming_url = repo_url + '/logstream'
+    import time
+    if provider:
+        streaming_url += ('/' + provider.lstrip('/'))
+
+    user, password = _get_site_credential(client, resource_group, name)
+    t = CurlThread(streaming_url, user, password)
+    t.daemon = True
+    t.start()
+
+    while True:
+        time.sleep(100) #so that ctrl+c can stop the command
+
+def download_historical_logs(resource_group, name, log_file=None, slot=None):
+    '''
+    Download historical logs as a zip file
+    '''
+    client = web_client_factory()
+    user, password = _get_site_credential(client, resource_group, name)
+    repo_url = _get_repo_url(client, resource_group, name, slot)
+    host = urlparse(repo_url).netloc
+    url = "https://{}:{}@{}/dump".format(user, password, host)
+    import requests
+    r = requests.get(url, stream=True)
+    with open(log_file, 'wb') as f:
+        for chunk in r.iter_content(chunk_size=1024):
+            if chunk: # filter out keep-alive new chunks
+                f.write(chunk)
+    return 'Done'
```

Comment:
Instead of returning `'Done'`, how about using the logger.
e.g.:`logger.warning('Downloaded logs to {}'.format(log_file))`
Then this method can return None as 'Done' is more of a status message rather than actual data that an be queried etc.